scripts/demo.sh

```
@@ -1,128 +1,408 @@
+pause
+
+# ════════════════════════════════════════════════════════════════
+# PART 4: Idempotency with Redis
+# ════════════════════════════════════════════════════════════════
+echo -e "${MAGENTA}╔════════════════════════════════════════════════════════════════╗${NC}"
+echo -e "${MAGENTA}║  Part 4: Idempotency Protection (Duplicate Prevention)        ║${NC}"
+echo -e "${MAGENTA}╚════════════════════════════════════════════════════════════════╝${NC}"
+echo ""
+
+echo -e "${CYAN}► First deposit request (£250):${NC}"
+IDEMPOTENCY_KEY="DEMO-$(date +%s)-$RANDOM"
```

Shellcheck (0.11.0)

[warning] IDEMPOTENCY_KEY appears unused. Verify use (or export if used externally).

(SC2034)